src/plugins/simulator/visualizations/qt-opengl/qtopengl_camera.cpp

```diff
@@ -67,308 +64,213 @@ namespace argos {
-                  if(it == m_listTimeline.end())
-                     m_listTimeline.push_back(psTimelineItem);
-                  else
-                     m_listTimeline.insert(it, psTimelineItem);
+               if(itTimelineElement->Value() == "keyframe") {
+                  GetNodeAttribute(*itTimelineElement, "placement", unKeyframePlacement);
+                  GetNodeAttribute(*itTimelineElement, "step", unKeyframeStep);
+                  if(unKeyframeStep < unLastStep) {
+                     THROW_ARGOSEXCEPTION("Keyframes must be in increasing order");
+                  }
+                  if(m_unLoop && unKeyframeStep > m_unLoop) {
+                     THROW_ARGOSEXCEPTION("Keyframe step must be less than the loop period");
+                  }
+                  else {
+                     unLastStep = unKeyframeStep;
+                     m_vecKeyframes.emplace_back(unKeyframeStep,
+                                                 unKeyframePlacement,
+                                                 false);
+                  }
+               }
+               else if(itTimelineElement->Value() == "interpolate") {
+                  if(!m_vecKeyframes.empty()) {
+                     m_vecKeyframes.back().InterpolateToNext = true;
+                  }
                }
-            }
-            for(std::list<STimelineItem*>::iterator it = m_listTimeline.begin();
-                it != m_listTimeline.end();
-                ++it) {
             }
          }
-         catch(CARGoSException& ex) {
-            THROW_ARGOSEXCEPTION_NESTED("Error initializing QTOpenGL camera placement", ex);
-         }
+         /* Reset the camera */
+         Reset();
+      }
+      catch(CARGoSException& ex) {
+         THROW_ARGOSEXCEPTION_NESTED("Error initializing QTOpenGL camera", ex);
       }
    }
 
    /****************************************/
    /****************************************/
 
    void CQTOpenGLCamera::Reset() {
-      m_unActivePlacement = 0;
+      if(m_bHasTimeline) {
+         m_bEnableTimeline = true;
+         UpdateTimeline();
+      }
+      else {
+         SetActivePlacement(0);
+      }
+   }
+
+   /****************************************/
+   /****************************************/
+
+   void CQTOpenGLCamera::Look() {
+      const CVector3& cPosition = GetActivePlacement().Position;
+      const CVector3& cTarget = GetActivePlacement().Target;
+      const CVector3& cUp = GetActivePlacement().Up;
+      gluLookAt(cPosition.GetX(),
+                cPosition.GetY(),
+                cPosition.GetZ(),
+                cTarget.GetX(),
+                cTarget.GetY(),
+                cTarget.GetZ(),
+                cUp.GetX(),
+                cUp.GetY(),
+                cUp.GetZ());
    }
 
    /****************************************/
    /****************************************/
 
    void CQTOpenGLCamera::Rotate(const QPoint& c_delta) {
```

Comment:
Why is rotate accepting a `QPoint`? A rotation should expect an angle...

Reply:
@ilpincy that was how it was when I started. The code has just just refactored, the logic should not have changed...

Reply:
The original function was meant to be a convenience function that used simpler functions to do its job. The real work is done in the simpler functions. I prefer the original structure because it's more readable.

Reply:
As I mentioned in another thread on this PR, I can revert this back to be more similar to how it was before.

Reply:
Feel free to go ahead! I'm just saying that I'll give this PR a thorough look on August 11 (working on a proposal) and I might change a couple of things, so the work you did is already a lot. I don't want to make you waste more time for aspects I could fix myself with a bit of copy-paste.